x-pack/filebeat/input/azureblobstorage/client.go

```diff
@@ -0,0 +1,102 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package azureblobstorage
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+
+	"github.com/elastic/beats/v7/x-pack/filebeat/input/azureblobstorage/types"
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+// fetchServiceClientAndCreds , generic function that returns a ServiceClient instance and a ServiceCredentials instance
+func fetchServiceClientAndCreds(cfg config, url string, log *logp.Logger) (*azblob.ServiceClient, *types.ServiceCredentials, error) {
+	if cfg.Auth.SharedCredentials != nil {
+		return fetchServiceClientWithSharedKeyCreds(url, cfg.AccountName, cfg.Auth.SharedCredentials, log)
+	} else if cfg.Auth.ConnectionString != nil {
+		return fetchServiceClientWithConnectionString(cfg.Auth.ConnectionString, log)
+	}
+
+	return nil, nil, fmt.Errorf("no valid auth specified")
+}
+
+func fetchServiceClientWithSharedKeyCreds(url string, accountName string, sharedKeycfg *sharedKeyConfig, log *logp.Logger) (*azblob.ServiceClient, *types.ServiceCredentials, error) {
+	// Creates a default request pipeline using your storage account name and account key.
+	credential, err := azblob.NewSharedKeyCredential(accountName, sharedKeycfg.AccountKey)
+	if err != nil {
+		log.Errorf("Invalid credentials with error: %v", err)
+		return nil, nil, err
+	}
+
+	serviceClient, err := azblob.NewServiceClientWithSharedKey(url, credential, nil)
```

Comment:
s/serviceClient/client/ to reduce weigh on the page (the type tell you that it's a service client). Similar below.